open-api/rest-catalog-open-api.yaml

```diff
@@ -3324,6 +3324,274 @@ components:
+      type: string
+      description:
+        "Timestamp type values follow the 'YYYY-MM-DDTHH:MM:SS.ssssss' ISO-8601 format with microsecond precision"
+      example: "2007-12-03T10:15:30.123456"
+
+    TimestampTzTypeValue:
+      type: string
+      description:
+        "TimestampTz type values follow the 'YYYY-MM-DDTHH:MM:SS.ssssss+00:00' ISO-8601 format with microsecond precision, 
+        and a timezone offset (+00:00 for UTC)"
+      example: "2007-12-03T10:15:30.123456+00:00"
+
+    TimestampNanoTypeValue:
+      type: string
+      description:
+        "Timestamp_ns type values follow the 'YYYY-MM-DDTHH:MM:SS.sssssssss' ISO-8601 format with nanosecond precision"
+      example: "2007-12-03T10:15:30.123456789"
+
+    TimestampTzNanoTypeValue:
+      type: string
+      description:
+        "Timestamp_ns type values follow the 'YYYY-MM-DDTHH:MM:SS.sssssssss+00:00' ISO-8601 format with nanosecond 
+        precision, and a timezone offset (+00:00 for UTC)"
+      example: "2007-12-03T10:15:30.123456789+00:00"
+
+
+    FixedTypeValue:
+      type: string
+      description:
+        "Fixed length type values are stored and serialized as an uppercase hexadecimal string 
+        preserving the fixed length"
+      example: "78797A"
+
+    BinaryTypeValue:
+      type: string
+      description:
+        "Binary type values are stored and serialized as an uppercase hexadecimal string"
+      example: "78797A"
+
+
+    PartitionData:
+      type: object
+      description:
+        "Partition data is serialized, where field id are preserved as a string JSON key, and the fields value is 
+        serialized based on the defined types"
+      additionalProperties:
+        oneOf:
+          - $ref: '#/components/schemas/PrimitiveTypeValue'
+      example:
+        {
+          "4": 1,
+          "5": "bar"
+        }
+
+    CountMap:
+      type: object
+      properties:
+        keys:
+          type: array
+          items:
+            $ref: '#/components/schemas/IntegerTypeValue'
+          description: "List of integer column ids for each corresponding value"
+        values:
+          type: array
+          items:
+            $ref: '#/components/schemas/LongTypeValue'
+          description: "List of Long values, matched to 'keys' by index"
+      example:
+        {
+          "keys": [ 3, 4 ],
+          "values": [ 100,200 ]
+        }
+
+    ValueMap:
+      type: object
+      properties:
+        keys:
+          type: array
+          items:
+            $ref: '#/components/schemas/IntegerTypeValue'
+          description: "List of integer column ids for each corresponding value"
+        values:
+          type: array
+          items:
+            $ref: '#/components/schemas/BinaryTypeValue'
+          description: "List of Binary values encoded as hexadecimal strings, matched to 'keys' by index"
+      example:
+        {
+            "keys": [ 3,4 ],
+            "values": [ "05000000","0A000000" ]
```

Comment:
Rather than encoding binary, I expected this to use `PrimitiveTypeValue` for keys.

Either way, the values need to be interpreted using the corresponding primitive type. The main tradeoffs are:
1. Readability and consistency in JSON
2. Plumbing the schema through to where ContentFile is serialized or deserialized

I think the stronger argument is for consistency. Why define all the types below only to use a binary representation here and an embedding for `PartitionData`?

Reply:
+1